content/blog/multitenancy-for-kubernetes-clusters-using-hpe-alletra-6000-and-nimble-storage.md

```diff
@@ -0,0 +1,234 @@
+
+YAML declarations below are created with:
+
+```markdown
+kubectl create -f-
+< Paste the YAML content >
+Hit CTRL-D on a new line.
+```
+
+Let's create a `Secret` referencing the HPE Alletra 6000 with the tenant credentials.
+
+```yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: hpe-backend
+  namespace: hpe-storage
+stringData:
+  serviceName: alletra6000-csp-svc
+  servicePort: "8080"
+  backend: 192.168.1.30
+  username: K8sAdminProd
+  password: qweqwe123
+```
+
+Next, create a new default `StorageClass`.
+
+```yaml
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+  name: hpe-standard
+provisioner: csi.hpe.com
+parameters:
+  csi.storage.k8s.io/fstype: xfs
+  csi.storage.k8s.io/controller-expand-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-expand-secret-namespace: hpe-storage
+  csi.storage.k8s.io/controller-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/controller-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  description: "Volume created by the HPE CSI Driver for Kubernetes"
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+At this point the Kubernetes cluster is ready to accept `PersistentStorageClaims`. Let's create one.
+
+```yaml
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: my-first-pvc
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 32Gi
+```
+
+Once the `PersistentVolume` has been bound we can inspect the array and determine that volume has been placed in the tenant's folder.
+
+![HPE Alletra 6000 Multitenancy for Kubernetes Clusters](/img/multitenancy-hpedev-screenshot.png "HPE Alletra 6000 Multitenancy for Kubernetes Clusters")
+
+That's it. Multitenant storage is now properly configured!
+
+# Container Storage Provider (CSP) access only
+
+Multitenant storage is only available today through the HPE Alletra 6000 and Nimble Storage Container Storage Provider for Kubernetes. The CSP uses an undisclosed REST API resource of the array to perform CRUD operations on objects that in turn are being tagged and grouped accordingly to create the notion of a full blown cloud experience both from the Kubernetes administrator perspective but most importantly for the storage administrator. The storage administrator does not need to worry about storage system credentials being compromised and wreaking havoc beyond the compartment they were assigned for.
```

Comment:
Add "(CSP)" after "Nimble Storage Container Storage Provider".

Second sentence: place commas around the phrase "in turn". Change the end of the sentence to read: "...both from the Kubernetes administrator perspective and, most importantly, for the storage administrator."